### PR TITLE
Add plane to stasis constraint and remove pressure constraints

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
### Description

Following the implementation of temperature-dependent solid domain constraint for both single-phase and two-phase flows, a need for an additional condition was identified. Indeed, in some simulations, splatters may happen, and with the previous implementation, splatter could freeze "mid-air" which is non-physical and therefore undesired. This PR adds a plane to divide the problem domain in two parts and only the one in the opposite direction to the normal vector is considered for stasis constraints to be applied. The plane is defined through a point and an outward pointing normal vector.

During this PR, a bug regarding the pressure constraints was detected and fixed. Previously, the pressure DOFs were also constrained, but this resulted in an ill-posed problem when sources terms in the momentum equation were pressure-dependent. Therefore, as of this PR, the constraints on pressure DOFs were removed.

### Testing

Previous tests with solid domain constraints were updated due to the pressure DOFs that are no more constrained. 
Additionally, two new application tests were added for the new plane restriction feature (1 for single fluid flows and one for VOF).

### Documentation

The "Constrain Stasis" page in the parameter documentation was updated with the new features and disabled ones.

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Lethe documentation is up to date
- [x] New feature has unit test(s) (preferred) or application test(s), and restart files are in the generator folder
- [x] The branch is rebased onto master
- [x] Changelog (CHANGELOG.md) is up to date
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent

Pull request related list:
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [x] If the fix is temporary, an issue is opened
- [x] The PR description is cleaned and ready for merge

### CODE REVIEW

**Review by oguevremont — COMMENTED** [review]
Good work!

File: doc/source/parameters/cfd/constrain_stasis.rst [hunk 1]
```diff
@@ -28,6 +32,12 @@ The subsection with default parameters goes as follows:
 
 * The ``enable`` parameter is set to ``true`` when at least one temperature-dependent stasis constraint should be applied.
 
+* The ``enable domain restriction with plane`` parameter is set to ``true`` when it is desired to reduce the constrained domain using a plane in addition to the temperature range. The plane cuts the domain in two subdomains. Only cells in the subdomain in the opposite direction to the normal vector can be constrained.
+
+* The ``restriction plane point`` parameter is a point on the plane.
+
+* The ``restriction plane normal vector`` parameter is an outward pointing normal vector of the plane. This vector is used to select which side of the plane contains cells to constrain.
```
[the remark this answers is not in the document]
Comment (blaisb) [reply doc/source/parameters/cfd/constrain_stasis.rst:39]: agreed with Olivier

**Review by blaisb — APPROVED** [review]
Other than Olivier's comment, I am fine with this Pr.

**Comment by blaisb** [comment]
can you rebase? Then I would merge afterwards

**Comment by blaisb** [comment]
I'll wait until @oguevremont and @hepap accept the PR before merging

File: include/solvers/navier_stokes_base.h [hunk 2]
```diff
@@ -482,9 +524,8 @@ class NavierStokesBase : public PhysicsSolver<VectorType>
   }
 
   /**
-   * @brief Flag cell DOFs (either "solid" or "connected to fluid") and
-   * constrain velocity DOFs with homogeneous constraints if cell is considered
-   * solid.
+   * @brief Indentify if temperature DOFs of the cell are within the
```
Comment (oguevremont) [inline include/solvers/navier_stokes_base.h:527]:
```suggestion
   * @brief Identify if temperature DOFs of the cell are within the
```

**Review by oguevremont — APPROVED** [review]

**Review by hepap — APPROVED** [review]
Looks very good to me :) Can't wait to try it!

**Pull request merged**